src/core/queries.ml

```diff
@@ -98,14 +111,22 @@ let handle_query : Sig_state.t -> proof_state option -> p_query -> unit =
       let h = get_pp_hint sym in
       if h <> Unqual then out 1 " [%a]" pp_hint h;
       (match !(sym.sym_def) with
-      | Some t -> out 1 " ≔ %a\n" pp_term t
+      | Some t ->
+          out 1 " ≔ %a\n" pp_term t;
+          Some (Format.asprintf " ≔ %a" pp_term t)
       | None ->
           out 1 "\n";
-          List.iter (fun r -> out 1 "%a\n" pp_rule (sym, r)) !(sym.sym_rules))
+          List.iter (fun r -> out 1 "%a\n" pp_rule (sym, r)) !(sym.sym_rules);
```

Comment:
I realize that, to avoid a useless List.iter if nothing needs to be printed, it would be better to change this line as well as follows:
```
let rule oc r = Format.fprintf oc "%a\n" pp_rule (sym, r) in
out 1 "%a" (List.pp rule "") !(sym.sym_rules)
```

Reply:
Sorry, I don't understand what you meant. What is `List.pp`?

Reply:
See  `lambdapi/src/lplib/list.ml` to find the function `pp`. ;)

Reply:
Oh. Thanks!